information/specification.md

```diff
@@ -0,0 +1,826 @@
+      [accumulated coin selection](#accumulated-coin-selection).
+
+ * **Step 4**
+
+   If the _total value_ of the _removed UTxO entries_ is greater than the
+   value of the _removed output_, generate a coin whose value is equal to
+   the exact difference, and add it to the _change values_ field of the
+   [accumulated coin selection](#accumulated-coin-selection).
+
+ * **Step 5**
+
+   If the _unpaid output list_ is empty, **terminate** here.
+
+   Otherwise, return to **Step 1**.
+
+### Termination
+
+The algorithm terminates _successfully_ if the [available UTxO
+list](#available-utxo-list) is not depleted before the [unpaid output
+list](#unpaid-output-list) can be fully depleted (i.e., if all the outputs have
+been paid for).
+
+The [accumulated coin selection](#accumulated-coin-selection) is returned
+to the caller as the [coin selection](#coin-selection) result.
+
+The [available UTxO list](#available-utxo-list) is returned to the caller
+as the [remaining UTxO set](#remaining-utxo-set) result.
+
+## Random-Improve
+
+The **Random-Improve** coin selection algorithm works in _two phases_.
+
+In the _first_ phase, the algorithm iterates through each of the [requested
+outputs](#requested-output-list) in _descending order of coin value_, from
+_largest_ to _smallest_. For each output, the algorithm repeatedly selects
+entries at **random** from the [initial UTxO set](#initial-utxo-set), until the
+_total value_ of selected entries is enough to pay for that ouput.
+
+In the _second_ phase, the algorithm attempts to **improve** upon each of the
+UTxO selections made in the previous phase, by conservatively expanding the
+selection made for each output, in order to generate improved change values.
+
+The name of the algorithm is taken from the two-phase process of selecting
+UTxO entries at **random**, and then attempting to **improve** upon the
+selections.
+
+### Motivating Principles
+
+There are several motivating principles behind the design of the algorithm.
+
+#### Principle 1: Dust Management
+
+The probability that random selection will choose dust entries from a UTxO
+set _increases_ with the proportion of dust in the set.
+
+Therefore, for a UTxO set with a large amount of dust, there's a high
+probability that a random subset will include a large amount of dust.
+
+Over time, selecting entries randomly in this way will tend to _limit_ the
+amount of dust that accumulates in the UTxO set.
+
+#### Principle 2: Change Management
+
+As mentioned in the [Goals](#goals) section, it is desirable that coin
+selection algorithms, over time, are able to create UTxO sets that have
+_useful_ outputs: outputs that will allow us to process future payments with a
+_reasonably small_ number of inputs.
+
+If for each payment request of value **v** we create a change output of
+_roughly_ the same value **v**, then we will end up with a distribution of
+change values that matches the typical value distribution of payment
+requests.
```

Comment:
This one maybe deserves a small example? 

"To give a concrete example: if Alice often buys bread for 1€, then her wallet will self-organize itself over time to contain a lot of 1€ coins."

Reply:
> This one maybe deserves a small example?
> 
> "To give a concrete example: if Alice often buys bread for 1€, then her wallet will self-organize itself over time to contain a lot of 1€ coins."

Done! Added in 3471486a2031a36a47d9dd42d12aad897c4027e2.